src/Core/src/Handlers/WebView/WebViewHandler.Windows.cs

```diff
@@ -279,20 +273,29 @@ void InitialCookiePreloadIfNecessary(string url)
 
 			if (cookies is not null)
 			{
-				var existingCookies = GetCookiesFromPlatformStore(url);
-				foreach (HttpCookie cookie in existingCookies)
+				var existingCookies = await GetCookiesFromPlatformStore(url);
+
+				if (existingCookies.Count == 0)
+					return;
+
+				foreach (CoreWebView2Cookie cookie in existingCookies)
 				{
+					// TODO Ideally we use cookie.ToSystemNetCookie() here, but it's not available for some reason check bac later
 					if (cookies[cookie.Name] is null)
-						myCookieJar.SetCookies(uri, cookie.ToString());
+						myCookieJar.SetCookies(uri,
+							new Cookie(cookie.Name, cookie.Value, cookie.Path, cookie.Domain)
+							{
+								Expires = DateTimeOffset.FromUnixTimeMilliseconds((long)cookie.Expires).DateTime,
```

Comment:
Question, AFAIK the expires of both cookies objects have the same data structure used:

* System.Net.Cookie: public DateTime Expires { get; set; }
* CoreWebView2Cookie: public DateTime Expires { get; set; }

Why are we doing that extra cast and method call? DataTime is a struct so it is passed by value, not by ref, so we get a new copy of it, am I mistaken?

Reply:
Nope, see internal discussion about it. Same as why we can't use `ToSystemNetCookie()` we're looking at this version of `CoreWebView2Cookie` for some reason: https://learn.microsoft.com/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2cookie?view=webview2-winrt-1.0.1587.40#expires